packages/babel-traverse/src/path/index.ts

```diff
@@ -217,20 +217,146 @@ const NodePath_Final = class NodePath {
   }
 };
 
-Object.assign(
-  NodePath_Final.prototype,
-  NodePath_ancestry,
-  NodePath_inference,
-  NodePath_replacement,
-  NodePath_evaluation,
-  NodePath_conversion,
-  NodePath_introspection,
-  NodePath_context,
-  NodePath_removal,
-  NodePath_modification,
-  NodePath_family,
-  NodePath_comments,
-);
+const methods = {
+  // NodePath_ancestry
+  findParent: NodePath_ancestry.findParent,
+  find: NodePath_ancestry.find,
+  getFunctionParent: NodePath_ancestry.getFunctionParent,
+  getStatementParent: NodePath_ancestry.getStatementParent,
+  getEarliestCommonAncestorFrom:
+    NodePath_ancestry.getEarliestCommonAncestorFrom,
+  getDeepestCommonAncestorFrom: NodePath_ancestry.getDeepestCommonAncestorFrom,
+  getAncestry: NodePath_ancestry.getAncestry,
+  isAncestor: NodePath_ancestry.isAncestor,
+  isDescendant: NodePath_ancestry.isDescendant,
+  inType: NodePath_ancestry.inType,
+
+  // NodePath_inference
+  getTypeAnnotation: NodePath_inference.getTypeAnnotation,
+  _getTypeAnnotation: NodePath_inference._getTypeAnnotation,
+  isBaseType: NodePath_inference.isBaseType,
+  couldBeBaseType: NodePath_inference.couldBeBaseType,
+  baseTypeStrictlyMatches: NodePath_inference.baseTypeStrictlyMatches,
+  isGenericType: NodePath_inference.isGenericType,
+
+  // NodePath_replacement
+  replaceWithMultiple: NodePath_replacement.replaceWithMultiple,
+  replaceWithSourceString: NodePath_replacement.replaceWithSourceString,
+  replaceWith: NodePath_replacement.replaceWith,
+  _replaceWith: NodePath_replacement._replaceWith,
+  replaceExpressionWithStatements:
+    NodePath_replacement.replaceExpressionWithStatements,
+  replaceInline: NodePath_replacement.replaceInline,
+
+  // NodePath_evaluation
+  evaluateTruthy: NodePath_evaluation.evaluateTruthy,
+  evaluate: NodePath_evaluation.evaluate,
+
+  // NodePath_conversion
+  toComputedKey: NodePath_conversion.toComputedKey,
+  ensureBlock: NodePath_conversion.ensureBlock,
+  ...(!process.env.BABEL_8_BREAKING && !USE_ESM
+    ? {
+        arrowFunctionToShadowed:
+          // workaround for rollup
+          // @ts-expect-error babel 7 only
+          NodePath_conversion[String("arrowFunctionToShadowed")],
+      }
+    : {}),
```

Comment:
Please keep this in a separate statement, otherwise TS will include it's type also in the Babel 8 build.